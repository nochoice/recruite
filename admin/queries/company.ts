import { gql } from '@keystone-6/core/admin-ui/apollo';

export const QUERY_COMPANY =  gql`
query GetSingleCompany($id: ID) {
    company(
     where: {id: $id}
   ) {
     id
     name
     logo {
       url
     }
     addresses {
       addressCountry
       addressLocality
       postalCode
       streetAddress
     }
     contacts {
       name
       surname
       id
       email
       phone
       description
     }
     positions (
       where:{active: {equals: true}}
     ){
       title
       id
       active
       contacts {
        id
        name
        surname
      }
     }
     }
   }
`;

export const QUERY_COMPANIES_OPTIONS = gql`{
    companies {
        name
        id
    }
}`