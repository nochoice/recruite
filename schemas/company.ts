import { list } from "@keystone-6/core";
import { checkbox, image, relationship, text, file} from "@keystone-6/core/fields";
import address_fields from './address_fields';


export default list({
    // Here are the fields that `User` will have. We want an email and password so they can log in
    // a name so we can refer to them, and a way to connect users to posts.
    fields: {
      name: text({
        validation: { isRequired: true },
      }),
      positions: relationship({ ref: 'Position.company', many: true }),
      addresses: relationship({ 
          ref: 'Address', 
          ui: {
            displayMode: 'cards',
            cardFields: ['addressCountry', 'addressLocality'],
            inlineEdit: { fields: ['addressCountry', 'addressLocality'] },
            linkToItem: false,
            inlineConnect: false,
            inlineCreate: { fields: ['addressCountry', 'addressLocality', 'postOfficeBoxNumber', 'streetAddress'] },
          },
          many: true 
      }),
      contacts: relationship({ 
        ref: 'Contact', 
        ui: {
          displayMode: 'cards',
          cardFields: ['name', 'surname'],
          inlineEdit: { fields: ['name', 'surname'] },
          linkToItem: false,
          inlineConnect: true,
          inlineCreate: { fields: ['name', 'surname'] },
        },
        many: true 
      }),
      contracts: file({ storage: 'company_contract',  }),
    },
    // Here we can configure the Admin UI. We want to show a user's name and posts in the Admin UI
    ui: {
      labelField: 'name',
      listView: {
        initialColumns: ['name', 'contacts'],
        defaultFieldMode: 'read'
      },
      itemView: { 
        defaultFieldMode: 'edit'
      },
      hideDelete: ({ session, context }) => {
        // console.log(session)
        return false
      },
      
      
    },
    description: "",
    

  });