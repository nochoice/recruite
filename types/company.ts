export interface Company {
    id: string;
    name: string;
    logo: {
        url: string
    }
}