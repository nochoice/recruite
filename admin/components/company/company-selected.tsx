
import { Avatar, AvatarGroup, Box, Card, CardContent, CardMedia, IconButton, Typography } from "@mui/material";
import { deepPurple } from "@mui/material/colors";
import React from "react";
import { Company } from "../../../types/company";
import { Contact } from "../../../types/contact";

interface Props {
    company: Company
}

export default function CompanySelected({company}: Props) {

    return (
        <Card>
            <CardContent>
                {company?.logo?.url && (
                    <CardMedia
                        component="img"
                        sx={{ height: 140 }}
                        image={company?.logo?.url}
                        alt={company.name}
                    />
                )}
                
                <Typography component="div" variant="h5" align="center">
                    {company.name} 
                </Typography>
            </CardContent>
        </Card>
    )
}

