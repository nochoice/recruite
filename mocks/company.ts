import { Company } from './../types/company';

export const MockCompany: Company = {
    id: "1",
    logo: {
        url: 'http://url.com'
    },
    name: 'No Choice'

}