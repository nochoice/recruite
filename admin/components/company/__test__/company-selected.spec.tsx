import React from 'react';
import { MockCompany } from '../../../../mocks/company';
import { render } from '../../../../tests/test-utils';
import CompanySelected from '../company-selected';

describe('CompanySelected', () => {
  it('should render', () => {
    const { container } = render(<CompanySelected company={MockCompany} />);
  
    expect(container).toMatchSnapshot();
  });

  it('should have right data', () => {
    const { getByText, getByTestId } = render(<CompanySelected company={MockCompany} />);
    
    expect(getByText(MockCompany.name)).toBeTruthy();
    expect(getByTestId('company-logo')).toBeTruthy();
  });

});
