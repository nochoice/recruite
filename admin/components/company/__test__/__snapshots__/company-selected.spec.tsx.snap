// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySelected should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <img
        alt="No Choice"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-pp913-MuiCardMedia-root"
        data-testid="company-logo"
        src="http://url.com"
      />
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
      >
        No Choice
      </div>
    </div>
  </div>
</div>
`;
